src/main/java/org/openstreetmap/atlas/generator/sharding/AtlasShardVerifier.java

```diff
@@ -56,6 +99,24 @@ protected int onRun(final CommandMap command)
     @Override
     protected SwitchList switches()
     {
-        return new SwitchList().with(ATLAS_FOLDER, EXPECTED_SHARDS, OUTPUT);
+        return new SwitchList().with(ATLAS_FOLDER, EXPECTED_SHARDS, OUTPUT, SPARK_OPTIONS,
+                LIST_DEPTH, PATH_FILTER_REGEX);
+    }
+
+    private Set<CountryShard> shardsFromFolder(final String expectedShardsPath,
+            final Map<String, String> sparkConfiguration, final int depth, final PathFilter filter)
+    {
+        logger.trace("listResourcesRecursively: {}", expectedShardsPath);
+        final Time start = Time.now();
+        final Set<CountryShard> result = FileSystemHelper
+                .streamPathsRecursively(expectedShardsPath, sparkConfiguration, filter, depth)
+                .map(Path::getName)
+                .map(name -> StringList.split(name, "/").last()
```

Comment:
this name should just be the atlas file name, correct? Not sure why we need to split on "/"

Reply:
So the map for the atlas file name is two lines below. This one is in case the reference comes from a folder, then the full paths will be included, and we want to strip them out here, if any! If the list comes from a reference file containing a list of shards to expect, then this map here will just do nothing

Reply:
Hadoop's Path.getName() returns just the filename, not the full path! So you can remove this extra map. https://hadoop.apache.org/docs/current/api/org/apache/hadoop/fs/Path.html

Reply:
Good catch! I will remove it